demos/mqtt/mqtt_demo_connection_sharing/mqtt_demo_connection_sharing.c

```diff
@@ -0,0 +1,1601 @@
+         * PUBLISH message on the precipitation topic. */
+        else if( globalReceivedPrecipitationData != true )
+        {
+            returnStatus = EXIT_FAILURE;
+            LogError( ( "Registered callback was not invoked on sending a PUBLISH to "
+                        "the precipitation topic: Topic=%s", DEMO_PRECIPITATION_TOPIC ) );
+        }
+    }
+
+    /* As we don't have any further PUBLISH/SUBSCRIBE operations in the loop iteration further
+     * we will perform clean-up operations of removing the subscription callbacks registrations,
+     * and unsubscribing from the temperature and humidity topic filters.*/
+
+    /* Remove all callbacks from the subscription manager. */
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        /* Remove the callback for temperature topics. */
+        SubscriptionManager_RemoveCallback( DEMO_TEMPERATURE_TOPIC_FILTER,
+                                            DEMO_TEMPERATURE_TOPIC_FILTER_LENGTH );
+
+        /* Remove the callback for the humidity topic filter. */
+        SubscriptionManager_RemoveCallback( DEMO_HUMIDITY_TOPIC,
+                                            DEMO_HUMIDITY_TOPIC_LENGTH );
+
+        /* Remove the callback for the precipitation topic filter. */
+        SubscriptionManager_RemoveCallback( DEMO_PRECIPITATION_TOPIC,
+                                            DEMO_PRECIPITATION_TOPIC_LENGTH );
+
+        /* Unsubscribe from the temperature topic filter. */
+        LogInfo( ( "Unsubscribing from the MQTT temperature topic filter: %.*s.",
+                   DEMO_TEMPERATURE_TOPIC_FILTER_LENGTH,
+                   DEMO_TEMPERATURE_TOPIC_FILTER ) );
+        returnStatus = unsubscribeFromTopic( &mqttContext,
+                                             DEMO_TEMPERATURE_TOPIC_FILTER,
+                                             DEMO_TEMPERATURE_TOPIC_FILTER_LENGTH );
+    }
+
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        /* Process Incoming UNSUBACK packet from the broker. */
+        mqttStatus = MQTT_ProcessLoop( &mqttContext, MQTT_PROCESS_LOOP_TIMEOUT_MS );
+
+        if( mqttStatus != MQTTSuccess )
+        {
+            returnStatus = EXIT_FAILURE;
+            LogError( ( "MQTT_ProcessLoop returned with status = %u.",
+                        mqttStatus ) );
+        }
+    }
+
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        /* Unsubscribe from the humidity topic. */
+        LogInfo( ( "Unsubscribing from the MQTT humidity topic filter %s.",
+                   DEMO_HUMIDITY_TOPIC ) );
+        returnStatus = unsubscribeFromTopic( &mqttContext,
+                                             DEMO_HUMIDITY_TOPIC,
+                                             DEMO_HUMIDITY_TOPIC_LENGTH );
+    }
+
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        /* Process Incoming UNSUBACK packet from the broker. */
+        mqttStatus = MQTT_ProcessLoop( &mqttContext, MQTT_PROCESS_LOOP_TIMEOUT_MS );
```

Comment:
It might save us some lines if `unsubscribeFromTopic` just processed the UNSUBACK for us. Or better yet, just call `unsubscribeFromTopic` three times in a row, then only call `MQTT_ProcessLoop` once

Reply:
Have updated the `unsubscribeFromTopic` to contain the call to `MQTT_ProcessLoop` to make the demo app logic cleaner.